src/components/DayPicker.jsx

```diff
@@ -88,6 +88,7 @@ const propTypes = forbidExtraProps({
   // month props
   renderMonthText: mutuallyExclusiveProps(PropTypes.func, 'renderMonthText', 'renderMonthElement'),
   renderMonthElement: mutuallyExclusiveProps(PropTypes.func, 'renderMonthText', 'renderMonthElement'),
+  renderWeekHeaderText: PropTypes.func,
```

Comment:
I like your suggestion of `renderWeekHeaderElement`, since an element would be a valid return value.